app/assets/javascripts/components/generic_object/custom-image-component.js

```diff
@@ -0,0 +1,89 @@
+ManageIQ.angular.app.component('customImageComponent', {
+  bindings: {
+    picture: '=',
+    newRecord: '<',
+    angularForm: '<',
+    pictureUrlPath: '@',
+    pictureUploaded: '=',
+    pictureRemove: '=',
+    pictureReset: '<',
+  },
+  controllerAs: 'vm',
+  controller: customImageComponentController,
+  templateUrl: '/static/generic_object/custom-image-component.html.haml',
+});
+
+customImageComponentController.$inject = ['$timeout'];
+
+function customImageComponentController($timeout) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.imageUploadStatus = "";
+    vm.changeImage = false;
+  };
+
+  vm.$onChanges = function() {
+    vm.changeImage = false;
+    restoreOriginalStatus();
+  };
+
+  vm.uploadClicked = function() {
+    var imageFile;
+
+    if (angular.element('#generic_object_definition_image_file')[0].files.length === 0) {
+      return;
+    }
+
+    imageFile = angular.element('#generic_object_definition_image_file')[0].files[0];
+
+    var reader = new FileReader();
+    vm.imageUploadStatus = "";
+
+    if (imageFile.type === 'image/png') {
+      vm.picture.extension = 'png';
+    } else if (imageFile.type === 'image/jpg') {
+      vm.picture.extension = 'jpg';
+    } else if (imageFile.type === 'image/jpeg') {
+      vm.picture.extension = 'jpeg';
+    } else {
+      vm.angularForm.generic_object_definition_image_file_status.$setValidity("incompatibleFileType", false);
+      vm.imageUploadStatus = __("Incompatible image type");
```

Comment:
Won't this show the message while the file is still loading? (or is the time really close to 0 even for huge images?)

Reply:
At this stage the file upload (or loading) hasn't even begun - it's just being vetted for the right extension.

Reply:
Yes.. but the file is being processed by the browser, in the background.

So, between you setting this and that `onload` event firing, the message will be visible.

Will test.. :)

Reply:
Oh.. sorry, I read the code wrong, no such problem here :). (I thought this happened always, not only on bad type.)

It does feel a bit weird that you have to punch the Upload button after chosing the file, but that's not a blocker, more of a detail for UX :).

Reply:
We are using the exact same UX in Catalog Explorer as well where we set custom images for Catalog Items. And that UX comes from `Bootstrap Filestyle` - http://markusslima.github.io/bootstrap-filestyle/

I did give a shot to "angularize"  Bootstrap Filestyle, as in, keep the Upload button disabled until a file is available to be uploaded, but once you run the JS script `$(":file").filestyle`, it pretty much disregards all angular directives etc

Reply:
> We are using the exact same UX in Catalog Explorer ..

Ah, ok, then definitely not a problem here :)

(Though just to be clear, my objection was not to what I see in bootstrap-filestyle, but to the extra button they have to press *after* they chose the file.)